### ISSUE TITLE
Empty SEARCH response is invalid according to RFC 3501

### ISSUE THREAD

**Issue by jonhoo** [greenmail-mail-test/greenmail#264]
If I sent `a1 SEARCH ALL` to GreenMail, it responds with `* SEARCH ` (note the trailing space). As @djc points out in https://github.com/djc/tokio-imap/issues/34#issuecomment-441171556, that's not technically a valid `SEARCH` reply, as it may only contain a trailing space if followed by at least one ID.

**Comment by jonhoo** [comment]
Specifically, I think what is happening here is that

https://github.com/greenmail-mail-test/greenmail/blob/e80650bb70e7eb0feee31b9cb92ddccf0b07f351/greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/SearchCommand.java#L86

ends up calling

https://github.com/greenmail-mail-test/greenmail/blob/ff620f83a994a904d04b16b8d8ddb02e7d1f8566/greenmail-core/src/main/java/com/icegreen/greenmail/imap/ImapResponse.java#L222-L223

and since `message = ""` (and not `null`), the space gets added. I also think

https://github.com/greenmail-mail-test/greenmail/blob/e80650bb70e7eb0feee31b9cb92ddccf0b07f351/greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/SearchCommand.java#L74-L76

shouldn't be necessary since `message()` ends up adding the space anyway.

I'm not sure if it's okay to make the check in `message()` also check for `!isEmpty()`, but a different workaround would be something along the lines of

```diff
- response.commandResponse(this, idList.toString());
+ if (uids.length == 0) {
+  response.commandResponse(this, null);
+ } else {
+  response.commandResponse(this, idList.toString());
+ }
```

**Comment by marcelmay** [comment]
Thx for the report and excellent analysis, @jonhoo.

 I'll go for the first approach fixing ```message()```. An empty message should not add whitespace.

**Comment by jonhoo** [comment]
Do you know roughly when 1.5.9 might be released along with this fix?

**Comment by marcelmay** [comment]
I'll do a release this week ~ Tuesday

**Comment by marcelmay** [comment]
Just released [1.5.9 ](https://github.com/greenmail-mail-test/greenmail/releases/tag/release-1.5.9)
